developer-lightspeed/compose-with-ollama.yaml

```
@@ -0,0 +1,42 @@
+services:
+  ollama:
+    image: docker.io/ollama/ollama:0.10.0-rc0
+    container_name: ollama
+    volumes:
+      - ${OLLAMA_MODELS_PATH:-ollama_data}:/root/.ollama
+    depends_on:
+      rhdh:
+        condition: service_started
+    network_mode: "service:rhdh"
+    command: >
+      "ollama serve & sleep 5 && ollama pull ${OLLAMA_MODEL:-tinyllama} && touch /tmp/ready && wait"
+    entrypoint: [ "sh", "-c" ]
+    restart: unless-stopped
+    healthcheck:
+      test: [ "CMD", "test", "-f", "/tmp/ready" ]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+  road-core-service:
+    image: quay.io/redhat-ai-dev/road-core-service:rcs-06302025-rhdh-1.6 # dclint disable-line service-image-require-explicit-tag
+    container_name: road-core-service
+    depends_on:
+      rhdh:
+        condition: service_started
+      ollama:
+        condition: service_healthy
+    volumes:
+      - ./developer-lightspeed/configs:/opt/app-root/src/developer-lightspeed/configs:Z
+    environment:
+      - PROJECT=rhdh
+      - RCS_CONFIG_FILE=/opt/app-root/src/developer-lightspeed/configs/extra-files/road-core-service-config.yaml
+      - OPENAI_API_KEY=${LIGHTSPEED_SERVER_TOKEN}
+      - RHDH_CONFIG_FILE=/opt/app-root/src/developer-lightspeed/configs/app-config/app-config.lightspeed.local.yaml
+    env_file:
+      - path: "./default.env"
+        required: true
+      - path: "./.env"
+        required: false
+    network_mode: "service:rhdh"
```

```suggestion
  road-core-service:
    extends:
      file: "./compose.yaml"
      service: road-core-service
    depends_on:
      rhdh:
        condition: service_started
      ollama:
        condition: service_healthy
```

It seems like the definition of this service is quite similar to the one in `developer-lightspeed/compose.yaml`, except on the `depends_on` field. So I'd suggest reducing duplication by leveraging the [`extends`](https://github.com/compose-spec/compose-spec/blob/main/spec.md#extends) directive. I've just tested it and it seems to be working.. WDYT?

---

The Docker compose not working nicely with `extends` feature, not sure what am I missing here.


Alternative option is to use override pattern by passing all the base files

`podman compose -f compose.yaml -d developer-lightspeed/compose.yaml -f developer-lightspeed/compose-with-ollama.yaml up -d`, downside is the command becomes very long :(

we could abstract these commands into a script or Makefile to provide simple commands for starting and stopping the services, eg:


 `make -C developer-lightspeed up`
 `make -C developer-lightspeed down`
 
 or 
 
 `./developer-lightspeed.sh up`
 `./developer-lightspeed.sh down`
 
 let me know what do you think about this?

---

Ah, indeed, looks like `podman-compose` and `docker-compose` do not behave the same way when resolving the relative file path for `extends`, as reported in https://github.com/containers/podman-compose/issues/703 for example.
And according to the [spec](https://github.com/compose-spec/compose-spec/blob/main/spec.md#finding-referenced-service), the path should be "relative to the location of the main Compose file", but this seems to interpreted differently :/

> Alternative option is to use override pattern by passing all the base files
>
> `podman compose -f compose.yaml -d developer-lightspeed/compose.yaml -f developer-lightspeed/compose-with-ollama.yaml up -d`, downside is the command becomes very long :(
> 

I agree that this option makes the command too long and may not be practical. I would personally prefer having to specify only a single compose file per "flavor" on top of the main compose file.

> we could abstract these commands into a script or Makefile to provide simple commands for starting and stopping the services

This would work for lighspeed, but I think users should ultimately be able to "merge" different compose files depending on their use cases; for example, if they want to have both orchestrator and lightspeed plugins enabled, they could just specify the corresponding compose files when running `{docker,podman} compose`. So starting with the native "podman compose" or "docker compose" command makes things easier IMO.

With that being said, I think we can revert the above suggestion about leveraging `extends` and go back to the duplicated services in both compose files for now. We'll need to investigate how to improve later?

---

Agreed, lets use override pattern for now. I have reverted this change.